elements/pl-symbolic-input/pl-symbolic-input.py

```diff
@@ -59,7 +67,17 @@ def render(element_html, data):
     imaginary_unit = pl.get_string_attrib(element, 'imaginary-unit-for-display', IMAGINARY_UNIT_FOR_DISPLAY_DEFAULT)
     size = pl.get_integer_attrib(element, 'size', SIZE_DEFAULT)
 
-    operators = ['cos', 'sin', 'tan', 'arccos', 'arcsin', 'arctan', 'acos', 'asin', 'atan', 'arctan2', 'atan2', 'exp', 'log', 'sqrt', '( )', '+', '-', '*', '/', '^', '**']
+    enable_more_func_names =  pl.get_boolean_attrib(element, 'enable-more-function-names', ENABLE_MORE_FUNCTION_NAMES_DEFAULT)
+    if enable_more_func_names:
+        phs.CONSTANTS_INSTANCE = phs.CONSTANTS_INSTANCE.enable_extended_function_names()
```

Comment:
@jonatanschroeder 
Yes, this could very well end up being problematic if custom defined function names are reused between consecutive `pl-symbolic-input` tags, or say, if the definitions of the functions wrt different tag instances were expected to change. Please see my [response above](https://github.com/PrairieLearn/PrairieLearn/pull/4334#discussion_r679888697) for what I grok is happening when each distinct tag (with its own set of configurations and parameters) gets parsed. 

A fix for this prospective problem is not going to be simple. It would require wrapping up all of the functions defined inside `pl-symbolic-input.py` into a class and keeping track of instances (each with its own customized copy of `_Constants()`) and then going back outward in the PL question handling code to make sure we are keeping track of all of this. I will try to run a few tests. So far, reasonable applications do not cause problems.

Reply:
Wouldn't passing a context as parameter (as you already do) be enough?